cmd/ctr/commands/run/run.go

```diff
@@ -20,6 +22,62 @@ import (
 	"github.com/urfave/cli"
 )
 
+// ContainerFlags are cli flags specifying container options
+var ContainerFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "config,c",
+		Usage: "path to the runtime-specific spec config file",
+	},
+	cli.StringFlag{
+		Name:  "checkpoint",
+		Usage: "provide the checkpoint digest to restore the container",
+	},
+	cli.StringFlag{
+		Name:  "cwd",
+		Usage: "specify the working directory of the process",
+	},
+	cli.StringSliceFlag{
+		Name:  "env",
+		Usage: "specify additional container environment variables (i.e. FOO=bar)",
+	},
+	cli.StringSliceFlag{
+		Name:  "label",
+		Usage: "specify additional labels (i.e. foo=bar)",
+	},
+	cli.StringSliceFlag{
+		Name:  "mount",
+		Usage: "specify additional container mount (ex: type=bind,src=/tmp,dest=/host,options=rbind:ro)",
+	},
+	cli.BoolFlag{
+		Name:  "net-host",
+		Usage: "enable host networking for the container",
+	},
+	cli.BoolFlag{
+		Name:  "read-only",
+		Usage: "set the containers filesystem as readonly",
+	},
+	cli.StringFlag{
+		Name:  "runtime",
+		Usage: "runtime name (io.containerd.runtime.v1.linux, io.containerd.runtime.v1.windows, io.containerd.runtime.v1.com.vmware.linux)",
+		Value: fmt.Sprintf("io.containerd.runtime.v1.%s", runtime.GOOS),
+	},
+	cli.BoolFlag{
+		Name:  "tty,t",
+		Usage: "allocate a TTY for the container",
+	},
+}
+
+func loadSpec(path string, s *specs.Spec) error {
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return errors.New("cannot load spec config file")
+	}
+	if err := json.Unmarshal(raw, s); err != nil {
```

Comment:
this should now probably get the new json import for better JSON performance? (#2068)

But I'm fine if that is a follow-up PR as this is just client side, or if someone claims it isn't that important for client-side that's fine as well.

Reply:
don't change this.  I'm reverting the changes after doing more testing with the whole stack